src/provisioning-ofono.c

```diff
@@ -1025,31 +1071,38 @@ static void check_interfaces(struct modem_data *modem, DBusMessageIter *iter)
 	if (has_gprs) {
 		if (get_gprs_properties(modem) < 0)
 			clean_provisioning();
-	} else
-		clean_provisioning();
+	} else {
+		LOG("Provisioning failed for modem path: %s",modem->path);
+	}
 
 }
-static void remove_modem()
+static void remove_modems()
```

Comment:
It can be called clean-up, it can be called destroy, but whatever you call it, it's supposed to free the (modem) data being removed and NOTHING ELSE.

https://developer.gnome.org/glib/stable/glib-Hash-Tables.html